peer/sync.go

```diff
@@ -82,20 +82,23 @@ func (syncer *Syncer) Sync(ctx context.Context, contentID []byte, hint *id.Signa
 	}
 	syncer.pendingMu.Unlock()
 
+	// Allow synchronisation messages for the content ID. This is required in
+	// order for channel to not filter inbound content (of unknown size). At the
+	// end of the method, we Deny the content ID again, un-doing the Allow and
+	// blocking content again.
+	syncer.filter.Allow(contentID)
+	defer func() {
+		time.Sleep(syncer.opts.WiggleTimeout)
+		syncer.filter.Deny(contentID)
```

Comment:
This might need to be in a goroutine, otherwise calls to `Sync` will block for the `WiggleTimeout`.